client/dashboard/domains/domain-forwarding-form/index.tsx

```diff
@@ -0,0 +1,332 @@
+	const basicFieldIds = [ 'sourceType', 'subdomain', 'targetUrl' ];
+	const basicFields = fields.filter( ( field ) => basicFieldIds.includes( field.id ) );
+
+	const form = {
+		type: 'regular' as const,
+		fields: basicFieldIds,
+	};
+
+	const handleSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+
+		const { target_host, target_path, is_secure } = parseTargetUrl( formData.targetUrl );
+
+		const submitData: DomainForwardingFormData = {
+			subdomain:
+				formData.sourceType === 'subdomain' ? formData.subdomain.trim() || undefined : undefined,
+			target_host,
+			target_path,
+			is_secure,
+			is_permanent: formData.redirectType === 'permanent',
+			forward_paths: formData.pathForwarding === 'yes',
+		};
+
+		// Add ID if editing
+		if ( isEdit && existingForwarding ) {
+			submitData.domain_redirect_id = existingForwarding.domain_redirect_id;
+		}
+
+		saveMutation.mutate( submitData, {
+			onSuccess: () => {
+				createSuccessNotice(
+					isEdit
+						? __( 'Domain forwarding rule updated successfully.' )
+						: __( 'Domain forwarding rule created successfully.' ),
+					{ type: 'snackbar' }
+				);
+				router.navigate( {
+					to: domainForwardingsRoute.fullPath,
+					params: { domainName },
+				} );
+			},
+			onError: () => {
+				createErrorNotice(
+					isEdit
+						? __( 'Failed to update domain forwarding rule.' )
+						: __( 'Failed to create domain forwarding rule.' ),
+					{ type: 'snackbar' }
+				);
+			},
+		} );
+	};
+
+	const { isPending } = saveMutation;
+
+	/*
+	const renderNoticeForPrimaryDomain = () => {
+		if ( ! domain?.isPrimary || domain?.is_domain_only_site ) {
+			return;
+		}
+
+		const noticeText = __(
+			"This domain is your site's main address. You can forward subdomains or {{a}}set a new primary site address{{/a}} to forward the root domain."
+			{
+				components: {
+					a: <a href={ `/domains/manage/${ domain.domain }` } />,
+				},
+			}
+		);
+
+		return (
+			<div className="domain-forwarding-card-notice">
+				<Icon
+					icon={ info }
+					size={ 18 }
+					className="domain-forwarding-card-notice__icon gridicon"
+					viewBox="2 2 20 20"
+				/>
+				<div className="domain-forwarding-card-notice__message">{ noticeText }</div>
+			</div>
+		);
+	};
+	*/
+
+	return (
+		<PageLayout
+			size="small"
+			header={
+				<PageHeader
+					title={ isEdit ? __( 'Edit Domain Forwarding' ) : __( 'Add Domain Forwarding' ) }
+				/>
+			}
+		>
+			<Card>
+				<CardBody>
+					<form onSubmit={ handleSubmit }>
+						<VStack spacing={ 4 }>
+							<DataForm< FormData >
+								data={ formData }
+								fields={ basicFields }
+								form={ form }
+								onChange={ ( edits: Partial< FormData > ) => {
+									setFormData( ( data ) => ( { ...data, ...edits } ) );
+								} }
+							/>
+
+							<Panel>
+								<PanelBody title={ __( 'Advanced settings' ) } initialOpen={ false }>
+									<PanelRow>
+										<VStack style={ { width: '100%' } } spacing={ 6 }>
```

Comment:
Do we really need this custom style? I tried removing it and there doesn't seem to be any difference 🤔 

```suggestion
										<VStack spacing={ 6 }>
```

Reply:
Yep, that's leftovers from before when I tried to use only the `PanelBody`